staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go

```diff
@@ -92,18 +93,174 @@ const (
 	defaultPodLogsTimeout = 20 * time.Second
 )
 
+var defaultContainerNameFromRefSpecRegexp = regexp.MustCompile(`spec\.(?:initContainers|containers|ephemeralContainers){(.+)}`)
+
+// LogsFlags reflects the information that the CLI is gathering via flags.
+// They will be converted to PodLogOptions for runtime requirements.
+type LogsFlags struct {
+	// required objects
+	RESTClientGetter genericclioptions.RESTClientGetter
+	ResourceBuilder  *resource.Builder
+
+	// flags
+	AllContainers                bool
+	Container                    string
+	Follow                       bool
+	IgnoreLogErrors              bool
+	InsecureSkipTLSVerifyBackend bool
+	LimitBytes                   int64
+	MaxFollowConcurrency         int
+	Prefix                       bool
+	Previous                     bool
+	PodRunningTimeout            time.Duration
+	Selector                     string
+	Since                        time.Duration
+	SinceTime                    string
+	Tail                         int64
+	Timestamps                   bool
+
+	// internal items
+	containerNameSpecified bool
+	namespace              string
+
+	genericclioptions.IOStreams
+}
+
+// NewLogsFlags returns a default LogsFlags.
+func NewLogsFlags(f cmdutil.Factory, streams genericclioptions.IOStreams) *LogsFlags {
+	namespace, _, _ := f.ToRawKubeConfigLoader().Namespace()
```

Comment:
You don't need that value yet, and you're swallowing error, which is not acceptable.